lib/ansiblelint/rules/VariableHasSpacesRule.py

```diff
@@ -13,7 +13,7 @@ class VariableHasSpacesRule(AnsibleLintRule):
     tags = ['formatting']
     version_added = 'v4.0.0'
 
-    bracket_regex = re.compile("{{[^{' ]|[^ '}]}}")
+    bracket_regex = re.compile("{{[^{' -]|[^ '}-]}}")
```

Comment:
please use raw-string literals for regexps
```suggestion
    bracket_regex = re.compile(r"{{[^{' -]|[^ '}-]}}")
```

Reply:
I'm not sure why this regexp does this check. Can anyone explain it to me? It seems wrong (even the original one)

Reply:
As I understand we want to match everything that's starting with `{{` not followed by space or hyphen. Also we want to allow `{{ '{{' }}` so we need to allow single quotes after `{{` aswell. Same thing for `}}`. I'm not sure why we wouldn't want to match `{{{` or `}}}` though.

Maybe @awcrosby can tell us why it's needed?

Problem is, this actually allows doing `{{'asdas'}}` as `{{'` and `'}}` are not matched. Same for `{{-asassa-}}`. But the regex to capture that would be something like: `({{-(?! ))|({{'(?! ))|({{[^ '-])|([^' -]}})|((?<=[^ ])-}})|((?<=[^ ])'}})` and I'm not positive we want to do that.

I'm not sure if python does allow the regex to be split into multiple lines? Then we probably could introduce it with `(# comment)` to explain it. Basically like:

```
({{-(?! ))|(# match everything with {{- not followed by a space OR)
({{'(?! ))|(# match everything with {{' not followed by a space OR)
({{[^ '-])|(# match everything with {{ not followed by a space, single quotation mark or hyphen OR)
([^' -]}})|(# match everything with }} not preceded by a space, single quotation mark or hyphen OR)
((?<=[^ ])-}})|(# match everything with -}} not preceded by a space OR)
((?<=[^ ])'}})(# match everything with '}} not preceded by a space)
```

Reply:
It does work with `\` at EOL, though I'm still not sure, we want to do that.